pkg/frontend/querymiddleware/roundtrip.go

```diff
@@ -42,7 +42,7 @@ type Config struct {
 	CacheResults           bool   `yaml:"cache_results"`
 	MaxRetries             int    `yaml:"max_retries" category:"advanced"`
 	ShardedQueries         bool   `yaml:"parallelize_shardable_queries"`
-	CacheUnalignedRequests bool   `yaml:"cache_unaligned_requests" category:"advanced"`
+	CacheUnalignedRequests bool   `yaml:"cache_unaligned_requests" category:"advanced" doc:"hidden"` // TODO: Deprecated in Mimir 2.10.0, remove in Mimir 2.12.0
```

Comment:
[nit] I suggest to rename this field to `DeprecatedCacheUnalignedRequests`, so we signal it even better in the code.